setup/ci/gcb.md

```diff
@@ -0,0 +1,66 @@
+---
+layout: single
+title:  "Google Cloud Build"
+sidebar:
+  nav: setup
+---
+
+{% include toc %}
+
+Setting up [Google Cloud Build](https://cloud.google.com/cloud-build/) as a Continuous Integration (CI)
+system within Spinnaker enables triggering pipelines when builds complete.
+
+## Prerequisites
+
+You need to have a [Google Cloud Platform](https://cloud.google.com) project with the
+[Cloud Build API](http://console.cloud.google.com/apis/library/cloudbuild.googleapis.com) enabled.
+You can enable the API with the following `gcloud` command:
+
+```
+gcloud services enable cloudbuild.googleapis.com
+```
+
+## Configure Spinnaker to Listen for Google Cloud Build Pub/Sub Notifications
+
+Google Cloud Build sends [Build Notifications](https://cloud.google.com/cloud-build/docs/send-build-notifications)
+when the state of your build changes. You can create a pipeline trigger to invoke a pipeline based
+on the status of your build.
+
+1. Create a Subscription object for the `cloud-builds` topic in your project:
+
+    ```
+    PROJECT_ID=
+    SUBSCRIPTION_NAME=googleCloudBuilds
+
+    gcloud pubsub subscriptions create $SUBSCRIPTION_NAME --topic projects/$PROJECT_ID/topics/cloud-builds
+    ```
+
+2. Configure and deploy Halyard with your newly created subscription.
+
+    ```
+    hal config pubsub google subscription add $SUBSCRIPTION_NAME --project $PROJECT_ID --subscription-name $SUBSCRIPTION_NAME
+
+    hal config pubsub google enable
+
+    hal deploy apply
+    ```
+
+## Configure Your Pipeline Trigger
+
+1. In your Pipeline configuration, click the **Configuration** stage on the far left of the pipeline diagram.
+
+2. Click **Automated Triggers**.
+
+3. In the **Type** field, select `Pub/Sub`.
+
+4. In the **Pub/Sub System Type** field, select `google`.
+
+5. In the **Subscription Name** field, select your `$SUBSCRIPTION_NAME` value.
+
+6. In the **Attribute Constraints** field, enter `status` in the **Key**, and `SUCCESS` (all upper case) in the **Value** field.
+
+7. In the **Payload Constraints** field, you can enter any of the top-level fields from the
+[Build object documentation](https://cloud.google.com/cloud-build/docs/api/reference/rest/v1/projects.builds#resource-build)
+as the key, and a Java regular expression as the value. For example, to match on an image produced,
+you can use the key `images` and the value `gcr\.io/my-project-id/my-application:latest` (replacing
+`my-project-id` and `my-application` with appropriate values).
```

Comment:
[minor] Perhaps something like "To trigger only on builds that publish a container image with the `latest` tag to the `gcr.io/my-project/my-application` repository, add a Payload Constraint with key `images` and value `gcr\.io/my-project-id/my-application:latest` (note that the `.` needs to be escaped since the constraint value is interpreted as a regular expression)"?

/cc @dorbin

Reply:
👍

Reply:
I think several of these instructions need to be updated because of artifact support from @ezimanyi . Eric, could you take a quick pass at how these instructions _should_ read now that we have first class GCB artifact support? IIRC, you mentioned something the other day about adding a specific filter, but I don't remember what it was.